src/averell/utils.py

```diff
@@ -145,34 +165,44 @@ def get_word_features(features):
 
 
 def get_syllable_features(features):
+    """
+    Filter the syllable features of a poem
+    :param features: poem dict
+    :return: Syllable dict list
+    """
     all_words_features = get_word_features(features)
     all_syllable_features = []
-    for stanza in features["stanzas"]:
+    word_number = 0
+    for stanza_index, stanza in enumerate(features["stanzas"]):
         lines = stanza["lines"]
-        for line_index, words in enumerate(lines):
-            for word_index, word in enumerate(words["words"]):
-                for syllable in word["syllables"]:
-                    word_features = all_words_features[word_index]
-                    line = lines[line_index]
+        for line in lines:
+            line_number = int(line["line_number"])
+            words = line["words"]
+            for word_index, word in enumerate(words):
+                syllables = word["syllables"]
+                for syllable in syllables:
                     syllable_features = {
                         "syllable": syllable,
-                        "line_number": line["line_number"],
-                        "metrical_pattern": line["metrical_pattern"],
-                        "manually_checked": word_features["manually_checked"],
-                        "title": word_features["title"],
-                        "author": word_features["author"],
-                        "stanza_type": word_features["stanza_type"],
-                        "word_text": word_features["word_text"],
-                        "line_text": line[line_index]["line_text"],
-                        "stanza_number": word_features["stanza_number"]
+                        "line_number": line_number,
                     }
+                    word_features = all_words_features[word_number]
+                    syllable_features.update(word_features)
                     all_syllable_features.append(syllable_features)
+                word_number += 1
     return all_syllable_features
 
 
 def filter_features(features, corpus_index, granularity=None):
+    """
+    Select the granularity
+    :param features: Corpora poems dict
+    :param corpus_index: Corpus index to be filtered
+    :param granularity:
```

Comment:
No description of granularity, add please

Reply:
Done